src/dotty/tools/dotc/typer/Applications.scala

```diff
@@ -553,6 +553,16 @@ trait Applications extends Compatibility { self: Typer with Dynamic =>
       // a modified tree but this would be more convoluted and less efficient.
       if (proto.isTupled) proto = proto.tupled
 
+      // If some of the application's arguments are function literals without explicitly declared
+      // parameter types, and the expected type is a value type, relate the
```

Comment:
This says `and the expected type is a value type` but I don't see that constraint checked in the code below